src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { useDispatch } from 'react-redux';
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, Form } from '@edx/paragon';
+import { Help, Post } from '@edx/paragon/icons';
+
+import { hidePostEditor } from '../data';
+import messages from './messages';
+
+function DiscussionPostType({
+  value,
+  type,
+  description,
+  icon,
+}) {
+  return (
+    <Form.Check type="radio" value={value} className="d-flex border thin p-2 mr-2 my-0">
+      {/* <Form.Check.Input type="radio" className=""/> */}
+      <Form.Check.Label className="d-flex flex-column align-items-center">
+        {icon}
+        <span>{type}</span>
+        <span className="x-small">{description}</span>
+      </Form.Check.Label>
+    </Form.Check>
+  );
+}
+
+DiscussionPostType.propTypes = {
+  value: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+  icon: PropTypes.element.isRequired,
+};
+
+function PostEditor({ intl }) {
+  const dispatch = useDispatch();
+  const cancelAdding = () => dispatch(hidePostEditor());
+
+  return (
+    <Form className="mx-4 my-2">
+      <h3>{intl.formatMessage(messages.heading)}</h3>
+      <Form.Group>
+        <Form.CheckboxSet className="d-flex flex-row my-3">
+          <DiscussionPostType
+            value="discussion"
+            type={intl.formatMessage(messages.discussionType)}
+            icon={<Post />}
+            description={intl.formatMessage(messages.discussionDescription)}
+          />
+          <DiscussionPostType
+            value="question"
+            type={intl.formatMessage(messages.questionType)}
+            icon={<Help />}
+            description={intl.formatMessage(messages.questionDescription)}
+          />
+        </Form.CheckboxSet>
+      </Form.Group>
+
+      <Form.Group>
+        <Form.Control
+          as="select"
+          defaultValue="General"
+          aria-describedby="topicAreaInput"
+          floatingLabel={intl.formatMessage(messages.topicArea)}
+        >
+          {/* TODO: topics has to be filled in another PR */}
+          <option>General</option>
+        </Form.Control>
+      </Form.Group>
+
+      <Form.Group>
+        <Form.Control
+          type="text"
+          aria-describedby="titleInput"
+          floatingLabel={intl.formatMessage(messages.postTitle)}
+        />
+      </Form.Group>
```

Comment:
Dropdown input field should have 50% width of the form and there is spacing issue between fields. There should be a section separator/ border between the dropdown and post-input field. 

Figma design:
<img width="443" alt="Screenshot 2021-08-10 at 2 22 08 PM" src="https://user-images.githubusercontent.com/79941147/128848489-473d0d18-7afe-41c9-9e0b-2ef5bc2e56d9.png">

Current design:
<img width="695" alt="Screenshot 2021-08-10 at 3 07 01 PM" src="https://user-images.githubusercontent.com/79941147/128848839-4b9421db-fa34-4249-ac09-475c34f495b4.png">